jgiven-tests/src/test/java/com/tngtech/jgiven/impl/GuaranteedStateTest.java

```diff
@@ -0,0 +1,51 @@
+package com.tngtech.jgiven.impl;
+
+import com.tngtech.jgiven.Stage;
+import com.tngtech.jgiven.annotation.BeforeStage;
+import com.tngtech.jgiven.annotation.ProvidedScenarioState;
+import com.tngtech.jgiven.exception.JGivenMissingGuaranteedScenarioStateException;
+import com.tngtech.jgiven.junit.ScenarioTest;
+import org.junit.Test;
+
+
+public class GuaranteedStateTest extends ScenarioTest<GuaranteedStateTest.GuaranteedGivenTestStage,
+                                                    GuaranteedStateTest.GuaranteedWhenThenTestStage,
+                                                    GuaranteedStateTest.GuaranteedWhenThenTestStage> {
+
+    @Test(expected = JGivenMissingGuaranteedScenarioStateException.class)
+    public void assure_before_method_of_second_test_is_executed_after_guaranteed_fields_validation() {
+        given().a_given_stage_with_guaranteed_field_uninitialized();
+        then().the_before_is_not_called();
+    }
+
+    @Test(expected = ClassNotFoundException.class)
+    public void assure_before_method_of_second_test_is_executed_if_guaranteed_initialized() {
+        given().a_given_stage_with_guaranteed_field_initialized();
+        then().the_before_is_not_called();
+    }
+
```

Comment:
Not a fully JGiven test because I find it very hard to check the order of execution without throwing these exceptions.

Reply:
Something like that could work. You'd have to check the `WhenTestFramework` stage for details